Server/Controllers/TransactionController.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using AutoMapper;
+using Endava_Project.Server.Data;
+using Endava_Project.Server.Models;
+using Endava_Project.Shared;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace Endava_Project.Server.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class TransactionController : ControllerBase
+    {
+        private readonly ApplicationDbContext context;
+        private readonly UserManager<ApplicationUser> userManager;
+
+        public TransactionController(ApplicationDbContext context, UserManager<ApplicationUser> userManager)
+        {
+            this.context = context;
+            this.userManager = userManager;
+        }
+
+        [HttpGet]
+        [Route("{typeFilter}/{sortFilter}/{orderFilter}/{itemsPerPage}/{pageNumber}")]
+        public TransactionHistoryData GetTransactions(string typeFilter,string sortFilter,string orderFilter, int itemsPerPage, int pageNumber)
+        {
+            var transactionData = new TransactionHistoryData { 
+                TransactionsList = new List<TransactionDto>()
+            };
+            var transactionsList = new List<Transaction>();
+            var userId = userManager.GetUserId(User);
+            var idList = context.Users.Include(e => e.Wallets).FirstOrDefault(x => x.Id == userId).Wallets.Select(e => e.Id).ToList();
```

Comment:
Would it make sense to have walletIds instead of IdList?

Reply:
In general i want to change the finding of transaction by wallets id to userid , beacause if a user will delet the wallet , he will not be able to see transaction made with that wallet . What do you think .

Reply:
Good point

Reply:
But here, I was thinking to rename idList to walletIds or something. To me, it's not quite clear what these are Ids are for. I had to analyze the query first

Reply:
yes , i need to work more on variables name.